src/objects.c

```diff
@@ -212,13 +213,20 @@ static CK_RV obj_add_to_pool(P11PROV_OBJ *obj)
     (void)MUTEX_UNLOCK(pool);
     /* ------------- LOCKED SECTION */
 
+    if (ret == CKR_OK) {
+        P11PROV_debug(
+            "Object added to pool (idx=%d, first_free=%d, obj=%p, num=%d)", idx,
```

Comment:
nevermind, I saw only later idx is used here.

Reply:
however you can simply use obj->poolid here, instead of extracting idx all they way out there, can you do that ?

Reply:
yeah it's probably better not to extract it. changed.